src/main/webapp/app/exam/manage/exams/exam-exercise-import/exam-exercise-import.component.ts

```diff
@@ -205,10 +235,75 @@ export class ExamExerciseImportComponent implements OnInit {
             // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
             exercise.shortName?.length! > 2 &&
             this.shortNamePattern.test(exercise.shortName!) &&
+            !this.exercisesWithDuplicatedShortNames.has(exercise.id!) &&
             exercise.shortName !== this.getBlocklistShortNameOfProgrammingExercise(exercise.id!)
         );
     }
 
+    /**
+     * checks if there are any selected exercises with the same title or short name as the passed exercise
+     * @param exercise     exercise we want to use to check for duplications
+     * @param exerciseGroup exercise group of the exercise
+     * @param checkForTitle true if the title should be checked, otherwise the short name is checked
```

Comment:
```suggestion
     * @param exercise      exercise we want to use to check for duplications
     * @param exerciseGroup exercise group of the exercise
     * @param checkForTitle true if the title should be checked, otherwise the short name is checked
```